tests/node-selector_test.go

```diff
@@ -0,0 +1,327 @@
+
+			// Checking the chaos operator running status
+			By("[Status]: Checking chaos operator status")
+			err = pkg.OperatorStatusCheck(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Operator status check failed, due to {%v}", err)
+
+			// Prepare Chaos Execution
+			By("[Prepare]: Prepare Chaos Execution")
+			err = pkg.PrepareChaos(&testsDetails, &chaosExperiment, &chaosEngine, clients, false)
+			Expect(err).To(BeNil(), "fail to prepare chaos, due to {%v}", err)
+
+			//Checking runner pod running state
+			By("[Status]: Runner pod running status check")
+			err = pkg.RunnerPodStatus(&testsDetails, testsDetails.AppNS, clients)
+			Expect(err).To(BeNil(), "Runner pod status check failed, due to {%v}", err)
+
+			// Chaos pod running status check
+			err = pkg.ChaosPodStatus(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Chaos pod status check failed, due to {%v}", err)
+
+			// Waiting for chaos pod to get completed
+			// And Print the logs of the chaos pod
+			By("[Status]: Wait for chaos pod completion and then print logs")
+			err = pkg.ChaosPodLogs(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Fail to get the experiment chaos pod logs, due to {%v}", err)
+
+			// Checking the chaosresult verdict
+			By("[Verdict]: Checking the chaosresult verdict")
+			chaosResult, err := pkg.GetChaosResultVerdict(&testsDetails, clients)
+			Expect(err).To(BeNil(), "ChasoResult Verdict check failed, due to {%v}", err)
+			Expect(chaosResult).To(Equal("Fail"), "ChaosResult verdict is not set to Fail, due to {%v}", err)
+
+			// Checking chaosengine verdict
+			By("Checking the Verdict of Chaos Engine")
+			chaosEngineVerdict, err := pkg.GetChaosEngineVerdict(&testsDetails, clients)
+			Expect(err).To(BeNil(), "ChaosEngine Verdict check failed, due to {%v}", err)
+			Expect(chaosEngineVerdict).To(Equal("Fail"), "ChaosEngine Verdict is not set to Fail, due to {%v}", err)
+		})
+	})
+
+	Context("Test for target node selection by node label", func() {
+		It("Executes the node-cpu-hog experiment using a valid NODE_LABEL value", func() {
+			testsDetails := types.TestDetails{}
+			clients := environment.ClientSets{}
+			chaosEngine := v1alpha1.ChaosEngine{}
+			chaosExperiment := v1alpha1.ChaosExperiment{}
+
+			//Getting kubeConfig and Generate ClientSets
+			By("[PreChaos]: Getting kubeconfig and generate clientset")
+			err := clients.GenerateClientSetFromKubeConfig()
+			Expect(err).To(BeNil(), "Unable to Get the kubeconfig, due to {%v}", err)
+
+			testsDetails.ExperimentName = "node-cpu-hog"
+
+			//Fetching all the default ENV
+			//Note: please don't provide custom experiment name here
+			By("[PreChaos]: Fetching all default ENVs")
+			klog.Infof("[PreReq]: Getting the ENVs for the %v test", testsDetails.ExperimentName)
+			environment.GetENV(&testsDetails, "node-cpu-hog", "node-selector-engine-2")
+
+			// Set TargetNodes to an empty string in order to use NodeLabel
+			testsDetails.TargetNodes = ""
```

Comment:
this can we do in environment.go?

Reply:
The experiment does expect a valid value of TARGET_NODES to be provided for the experiment run, here it is purposely being set as an empty string to meet the particular conditions of the specific test-case.

Reply:
can we set/initialise this variable from environment.go? If we need some other specific value then we can add here?

Reply:
Fixed